uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -269,49 +273,94 @@ async def send(self, message: "ASGISendEvent") -> None:
         message_type = message["type"]
 
         if not self.handshake_complete:
-            if message_type == "websocket.accept":
-                message = typing.cast("WebSocketAcceptEvent", message)
-                self.logger.info(
-                    '%s - "WebSocket %s" [accepted]',
-                    self.scope["client"],
-                    get_path_with_query_string(self.scope),
-                )
-                subprotocol = message.get("subprotocol")
-                extra_headers = self.default_headers + list(message.get("headers", []))
-                extensions: typing.List[Extension] = []
-                if self.config.ws_per_message_deflate:
-                    extensions.append(PerMessageDeflate())
-                if not self.transport.is_closing():
-                    self.handshake_complete = True
-                    output = self.conn.send(
-                        wsproto.events.AcceptConnection(
-                            subprotocol=subprotocol,
-                            extensions=extensions,
-                            extra_headers=extra_headers,
+            if not self.response_started:
```

Comment:
Can't we do the same as suggested for `websockets` i.e. remove this line from accept and close?